website/docs/r/nsxt_distributed_firewall_rule.html.markdown

```diff
@@ -0,0 +1,117 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_distributed_firewall_rule"
+sidebar_current: "docs-vcd-resource-nsxt-distributed-firewall-rule"
+description: |-
+  The Distributed Firewall allows user to segment organization virtual data center entities, such as
+  virtual machines, based on virtual machine names and attributes. 
+---
+
+# vcd\_nsxt\_distributed\_firewall\_rule
+
+The Distributed Firewall allows user to segment organization virtual data center entities, such as
+virtual machines, based on virtual machine names and attributes. 
+
+~> There is a different resource
+[`vcd_nsxt_distributed_firewall`](/providers/vmware/vcd/latest/docs/resources/nsxt_distributed_firewall)
+that can manage all firewall rules in one resource. One should use **only one of**
+`vcd_nsxt_distributed_firewall` or `vcd_nsxt_distributed_firewall_rule` as they will **override each
+other**.
+
+## Example Usage
+
+```hcl
+data "vcd_vdc_group" "existing" {
+  org  = "my-org" # Optional, can be inherited from Provider configuration
+  name = "main-vdc-group"
+}
+
+data "vcd_nsxt_network_context_profile" "cp1" {
+  context_id = vcd_vdc_group.test1.id
+  name       = "CTRXICA"
+  scope      = "SYSTEM"
+}
+
+resource "vcd_nsxt_distributed_firewall_rule" "r1" {
+  org          = "my-org"
+  vdc_group_id = vcd_vdc_group.test1.id
+
+  name        = "rule1"
+  action      = "ALLOW"
+  description = "description"
+
+  source_ids           = [vcd_nsxt_ip_set.set1.id, vcd_nsxt_ip_set.set2.id]
+  destination_ids      = [vcd_nsxt_security_group.g1-empty.id, vcd_nsxt_security_group.g2.id]
+  app_port_profile_ids = [vcd_nsxt_app_port_profile.p1.id, data.vcd_nsxt_app_port_profile.WINS.id, data.vcd_nsxt_app_port_profile.FTP.id]
+}
+
+resource "vcd_nsxt_distributed_firewall_rule" "r2" {
+  org          = "{{.Org}}"
+  vdc_group_id = vcd_vdc_group.test1.id
+
+  # Specifying a particular ID of other firewall rule will ensure that the current one is placed above
```

Comment:
I think this is in reverse. Current rule will go below the above_rule_id, won't it?

Reply:
No. The goal would be for it to go `above` the rule that is specified in `above_rule_id`. This is inspired by NSX-V rule, that had such field in API - https://registry.terraform.io/providers/vmware/vcd/latest/docs/resources/nsxv_firewall_rule#above_rule_id

It makes sense to put "above", because if it was places "below" - one would have no mechanism to place a new rule at the very top.